adoc/Private-Registry-Administration.adoc

```diff
@@ -743,3 +743,206 @@ suse-registry-harbor-registry-65854df7bc-mrfnj     2/2     Running   0
 suse-registry-harbor-trivy-0                       1/1     Running   1          24h
 suse-registry-harbor-trivy-1                       1/1     Running   0          24h
 ----
+
+=== Backup and Restore
+
+The backup and restore for {spr} can be performed by link:https://documentation.suse.com/suse-caasp/4.2/html/caasp-admin/_backup_and_restore_with_velero.html[Velero]. Velero is an application that performs backup and restore, disaster recovery, and the migration of Kubernetes cluster resources and persistent volumes.
+
+When {spr} is configured to be deployed together with its internal services (database, Redis and filesystem storage for image and artifacts) Velero is able to fully cover the backup and restore for it. However, as Velero is only responsible for Kubernetes resources, when {spr} is deployed using external service providers, such as managed PostgreSQL database or object storage (Amazon S3 or Azure Blob Storage), additional actions must be performed to ensure the backup of data that is external to the Kubernetes cluster.
+
+In general, the backup and restore process for {spr} can be split in two parts:
+
+* Backup and restore of Kubernetes resources (performed by Velero):
+** Namespace
+** Deployment
+** ReplicaSet
+** StatefulSet
+** Endpoint
+** Service
+** Ingress
+** ConfigMap
+** Secret
+** PersistentVolumeClaim
+** PersistentVolume
+** Pod
+** Other CRDs (e.g. `Certificate` when using `cert-manager` or `RedisFailover` when using `redis-operator`)
+* Backup and restore of external services:
+** Database
+** Redis
+** Object Storage
+
+==== Backup and restore of Kubernetes resources
+
+The following steps describe how to perform the backup and restore of Kubernetes resources using Velero.
+
+===== Backup
+
+. Install Velero as described in the link:https://documentation.suse.com/suse-caasp/4.2/html/caasp-admin/_backup_and_restore_with_velero.html[documentation]
+
+. (optional) If you are using a volume provider that does not support volume snapshots or a volume-snapshot API, or does not have Velero-supported storage plugin, Velero leverages link:https://restic.net/[restic] as a generic solution to backup and restore this sort of persistent volume. To perform a Restic backup of persistent volumes, Velero requires the addition of a specific annotation (`backup.velero.io/backup-volumes=<VOLUME_NAME_1>,<VOLUME_NAME_2>,...`) to the pods that have the volumes mounted. This can be achieved by adding the following entries to the {spr} helm chart `harbor-values.yaml`, and then performing `helm upgrade` to apply the annotations:
```

Comment:
```suggestion
. (optional) If you are using a volume provider that does not support volume snapshots or a volume-snapshot API, or does not have Velero-supported storage plugin, then Velero uses link:https://restic.net/[Restic] as a generic solution for backing up and restoring this sort of persistent volume. To perform a Restic backup of persistent volumes, Velero requires the addition of a specific annotation (`backup.velero.io/backup-volumes=<VOLUME_NAME_1>,<VOLUME_NAME_2>,...`) to the pods that have the volumes mounted. This can be achieved by adding the following entries to {spr}'s Helm chart (`harbor-values.yaml`), then performing `helm upgrade` to apply the annotations:
```